rate/fetchers/nbu_test.go

```diff
@@ -0,0 +1,28 @@
+package fetchers_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Hukyl/genesis-kma-school-entry/rate/fetchers"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNBUUnsupportedCurrency(t *testing.T) {
+	nbu := fetchers.NewNBURateFetcher()
+	_, err := nbu.FetchRate(context.Background(), "-", "UAH")
+	assert.Error(t, err)
+}
+
+func TestNBUFetchRate(t *testing.T) {
+	nbu := fetchers.NewNBURateFetcher()
+	rate, err := nbu.FetchRate(context.Background(), "USD", "UAH")
+	assert.NoError(t, err)
+	assert.Greater(t, rate.Rate, float32(0))
+}
+
+func TestNBUOnlyUAH(t *testing.T) {
+	nbu := fetchers.NewNBURateFetcher()
+	_, err := nbu.FetchRate(context.Background(), "USD", "EUR")
```

Comment:
why don't you add a test to check if your fetcher returns an error if the context's timeout has exceeded?

Reply:
and why don't combine these tests to table test again?